docs(l2): update based roadmap (#3319)
lambdaclass/ethrex: docs/l2/roadmap.md

| ------------------------------------------------------------------------------------- | -------------- | ----------------------------- | ------------------------------------------------- | -------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| [Taiko Alethia (Taiko Labs)](https://github.com/taikoxyz/taiko-mono)                  | Permissioned   | Fixed Deterministic Lookahead | Multi-proof (sgxGeth (TEE), and sgxReth (ZK/TEE)) | Yes      | -                                                                                                                                                                                                                                                                          |

> | [Gattaca's Based OP (Gattaca + Lambdaclass)](https://github.com/gattaca-com/based-op) | Permissioned   | Round Robin                   | Single Proof (optimistic)                         | Yes      | For phase 1, the Sequencer/Gateway is centralized. For phase 2 (current phase) the Sequencer/Gateway is permissioned.                                                                                                                                                      |

Reviewer: ```suggestion
| [Gattaca's Based OP (Gattaca + Lambdaclass)](https://github.com/gattaca-com/based-op) | Permissioned   | Round Robin                   | Single Proof (optimistic)                         | Yes      | For phase 1, the Sequencer/Gateway was centralized. For phase 2 (current phase), the Sequencer/Gateway is permissioned.                                                                                                                                                      |
```

> | [R1](https://ethereumr1.org/)                                                         | Permissionless | Total Anarchy                 | Multi-proof (ZK, TEE, Guardian)                   | No       | R1 is yet to be specified but planned to be built on top of Surge and Taiko's Stack. They're waiting until Taiko is mature enough to have preconfs                                                                                                                         |

Reviewer: ```suggestion
| [R1](https://ethereumr1.org/)                                                         | Permissionless | Total Anarchy                 | Multi-proof (ZK, TEE, Guardian)                   | No       | R1 is yet to be specified but plans are for it to be built on top of Surge and Taiko's Stack. They're waiting until Taiko is mature enough to have preconfs                                                                                                                         |
```

The following example assumes a sequencing period of 1 day, 1 auction challenge per hour with challenge periods of 1 hour.

![Diagram showing how challenges work](./img/challenges_to_leaders.png)

1. Auction winner (Sequencer green) starts as the lead Sequencer of the sequencing period.
2. No one can challenge the lead in the first hour.
3. During the second hour, the first auction challenge starts, and multiple Sequencers bid to challenge the lead. Finally, the lead Sequencer is overthrown and the new lead (Sequencer blue) starts sequencing.
4. In the third hour a new auction challenge opens and the former lead Sequencer takes back the lead.

> 5. Until the last hour of the sequencing period, the same cycle repeats having many changes of leaders.

Reviewer: ```suggestion
5. Until the last hour of the sequencing period, the same cycle repeats, having many leader changes.
```

- **Up-to-date Nodes:** Nodes that have the last committed batch in their storage and only miss the next batch.
- **Following:** We say that up-to-date nodes are **following** the lead Sequencer.
- **Syncing:** Nodes are **syncing** if they are not up-to-date. They’ll stop syncing after they reach the **following** state.
- **Verify Transaction:** An L1 transaction submitted by anyone to verify a ZK proof to an L2 batch execution.

### Network participants

- Sequencer Nodes: Nodes that have opted in to serve as Sequencers.
- Follower Nodes: State or RPC Nodes.
- Prover Nodes:

By default, every Ethrex L2 node begins as a Follower Node. A process will periodically query the L1 smart contract registry for the Lead Sequencer's address and update each node's state accordingly.

### Network parameters

A list of all the configurable parameters of the network.

- Sequencing period duration
- Auction period duration
- Number of sequencing periods in an allocated period
- Time between auction and allocated period
- L2 block time
- Minimum collateral in ETH for Sequencers registration
- Withdrawal delay for Sequencers that quit the protocol

> - Initial ticket auction price multiplier?

Author: [Remove question mark](https://github.com/lambdaclass/ethrex/pull/3319/commits/7752587ec8636793c6bd449b611aa60417adc685)
